app/src/test/java/org/simple/clinic/drugs/EditMedicineUpdateTest.kt

```diff
@@ -31,4 +43,48 @@ class EditMedicineUpdateTest {
             hasEffects(RefillMedicines(patientUuid)),
         ))
   }
+
+  @Test
+  fun `when filled prescribed drugs are fetched and some of the drugs are updated on the same day, then show set edit medicine button state to save button`() {
+    val model = EditMedicinesModel.create(patientUuid, RefillMedicine)
+    val updateSpec = UpdateSpec(EditMedicinesUpdate(LocalDate.of(2020, 11, 18), ZoneOffset.UTC))
+    val prescribedDrugRecords = listOf(
+        TestData.prescription(uuid = UUID.fromString("4aec376e-1a8f-11eb-adc1-0242ac120002"), name = "Amlodipine1", createdAt = LocalDate.of(2020, 11, 18).toUtcInstant(userClock), updatedAt = LocalDate.of(2020, 11, 18).toUtcInstant(userClock)),
+        TestData.prescription(uuid = UUID.fromString("537a119e-1a8f-11eb-adc1-0242ac120002"), name = "Amlodipine2", createdAt = LocalDate.of(2020, 11, 18).toUtcInstant(userClock), updatedAt = LocalDate.of(2020, 11, 18).toUtcInstant(userClock)),
+        TestData.prescription(uuid = UUID.fromString("5ac2a678-1a8f-11eb-adc1-0242ac120002"), name = "Amlodipine3", createdAt = LocalDate.of(2020, 11, 18).toUtcInstant(userClock), updatedAt = LocalDate.of(2020, 11, 18).toUtcInstant(userClock)),
+    )
+
+    val protocolDrugRecords = listOf(
+        TestData.protocolDrug(uuid = UUID.fromString("5f9f0fe2-1a8f-11eb-adc1-0242ac120002"), name = "Amlodipine4", createdAt = LocalDate.of(2020, 11, 18).toUtcInstant(userClock), updatedAt = LocalDate.of(2020, 11, 18).toUtcInstant(userClock))
+    )
+
+    val protocolDrugsAndDosages = listOf(ProtocolDrugAndDosages("Amlodipine5", protocolDrugRecords))
+
+    updateSpec
+        .given(model)
+        .whenEvent(DrugsListFetched(protocolDrugsAndDosages, prescribedDrugRecords))
+        .then(assertThatNext(
+            hasModel(model.prescribedDrugsFetched(prescribedDrugRecords).protocolDrugsFetched(protocolDrugsAndDosages).editMedicineDrugStateFetched(EditMedicineButtonState.SAVE_MEDICINE)),
+            hasNoEffects()
+        ))
+  }
+
+  @Test
+  fun `when empty prescribed drugs are fetched and none of the drugs are updated on the same day, then show set edit medicine button state to save button`() {
```

Comment:
Same comment as earlier about the name of this test being implementation specific.